cdc/generation.cc

```diff
@@ -294,6 +299,34 @@ future<db_clock::time_point> get_local_streams_timestamp() {
     });
 }
 
+// non-static for testing
+constexpr size_t limit_of_streams_in_topology_description() {
+    // Each stream takes 16B and we don't won't to exceed 1MB = 1024 * 1024 B
+    // so we can have at most 65536 streams but not less than 1 per vnode.
+    return 1024 * 1024 / 16;
+}
```

Comment:
Isn't `1MB` too strict? That will start truncating generations even for relatively small clusters. How about `4MB`? It will cause a little stall/allocation here and there, but should be harmless really, and it doesn't have the commitlog problem.

Reply:
Maybe you're right. It would start trimming clusters with 64cpus machines starting from 5th node and with 32cpus machines starting from 9th node. 4MB would give us 4 times that so 17th 64cpus node and 33rd 32cpus node. I will change the limit.

Reply:
Also maybe @avikivity will have some opinion on this